### PR TITLE
Rdf4j migration

### PR DESCRIPTION
**Pull request by elky87**
Hi, as discussed in our mails this is our migrated version of the original driver

### CODE REVIEW

**Comment by grechaw** [comment]
We've created a new repository for the rdf4j upgrade, which will be served from a new location.   
Thanks for submitting this PR for review, I'll take a look at the commits now.

File: marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/MarkLogicRepositoryConnectionTest.java [hunk 1]
```diff
@@ -1224,7 +1224,7 @@ public void testNestedConnections()
         config.setPort(port);
         config.setUser(user);
         config.setPassword(password);
-        config.setAuth("DIGEST");
+        config.setAuth("BASIC");
```
Comment (grechaw) [inline marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/MarkLogicRepositoryConnectionTest.java]: The swtich to BASIC is probably fine for this.  We have a long-standing requirement to support "DIGEST" though, and we've found that this is something that is not universally supported in HTTP clients.  Thus we will probably have to keep this "DIGEST" and see what issues, perhaps, you ran into.

File: marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/MarkLogicRepositoryCacheTest.java [hunk 2]
```diff
@@ -116,6 +116,7 @@ public void testSizeCommitWithWriteCache()
     {
         conn.setIsolationLevel(IsolationLevels.SNAPSHOT);
 
+        conn.clear();
```
Comment (grechaw) [inline marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/MarkLogicRepositoryCacheTest.java]: Is this new line to fix an issue or just good housekeeping?

File: marklogic-sesame-performance/build.gradle [hunk 3]
```diff
@@ -29,7 +30,7 @@ repositories {
 }
 
 dependencies {
-    compile('com.marklogic:marklogic-sesame:1.0.2.1-SNAPSHOT')
+    compile('com.marklogic:marklogic-sesame:2.0.0')
```
Comment (grechaw) [inline marklogic-sesame-performance/build.gradle]: Note to implementors that we have a new package name and will make a new version for the library that incorporates this work.

**Review by grechaw — COMMENTED** [review]
Thank you very much for this PR and for keeping us moving forward with our support for rdf4j.

The comments I have are very minor but do have a couple of questions for you.

All in all very straightforward and clear refactoring.

File: marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/query/MarkLogicGraphQueryTest.java [hunk 4]
```diff
@@ -130,6 +130,7 @@ public void testGraphQueryWithBaseURI()
         results.close();
     }
     @Test
+    @Ignore
```
Comment (elky87) [inline marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/query/MarkLogicGraphQueryTest.java]: Just a case of "lets look at that the next day" but then forgot to take a second look. Test is still failing.

File: marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/MarkLogicRepositoryConnectionTest.java (hunk 1 above)
Comment (elky87) [inline marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/MarkLogicRepositoryConnectionTest.java]: The reason for this switch was just my test infrastructure. I had a server that ran on BASIC Auth and thats why I made that switch.

File: marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/MarkLogicRepositoryConnectionTest.java [hunk 5]
```diff
@@ -515,7 +515,7 @@ public void testAddStatements() throws Exception{
         BooleanQuery booleanAliceQuery = conn.prepareBooleanQuery(QueryLanguage.SPARQL, checkAliceQuery);
         Assert.assertTrue(booleanAliceQuery.evaluate());
 
-        String checkBobQuery = "ASK { <http://example.org/people/bob> <http://example.org/ontology/age> '123123123123'^^<http://www.w3.org/2001/XMLSchema#double> .}";
+        String checkBobQuery = "ASK { <http://example.org/people/bob> <http://example.org/ontology/age> '1.23123123123E11'^^<http://www.w3.org/2001/XMLSchema#double> .}";
```
Comment (elky87) [inline marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/MarkLogicRepositoryConnectionTest.java]: I could not find any official statement on a behaviour change but doing the change made the "ask" work for me.

File: marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/MarkLogicRepositoryCacheTest.java (hunk 2 above)
Comment (elky87) [inline marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/MarkLogicRepositoryCacheTest.java]: I had some issue with old triples beeing leftover from a previous test thats why the line is there. But would be better to put it in the @before or @after  of the unit test

File: marklogic-sesame-performance/build.gradle (hunk 3 above)
Comment (elky87) [inline marklogic-sesame-performance/build.gradle]: Noted

File: marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/query/MarkLogicQueryTest.java [hunk 6]
```diff
@@ -29,15 +30,15 @@
...
+public class MarkLogicQueryTest extends SesameTestBase {
 
     public static DatabaseClient writerClient;
 
     protected static final String TESTFILE_OWL = "src/test/resources/testdata/test-small.owl";
 
     @Before
     public void setUp() throws FileNotFoundException {
-        writerClient = DatabaseClientFactory.newClient(host, port, user, password, DatabaseClientFactory.Authentication.DIGEST);
+        writerClient = DatabaseClientFactory.newClient(host, port, user, password, DatabaseClientFactory.Authentication.BASIC);
```
Comment (akshaysonvane) [inline marklogic-sesame/src/test/java/com/marklogic/semantics/sesame/query/MarkLogicQueryTest.java]: Noted.
```
DatabaseClientFactory.newClient(host, port, new DigestAuthContext(username, password));
```

**Comment by grechaw** [comment]
Thank you for these commits, @elky87

**Comment by lawrence-pressinnov** [comment]
hi @grechaw 
I can't find the new location for the marklogic-rdf4j project. I must miss something.
I am really interested in this upgrade (since #322) and would love to have a preview on it. 
Can you point me to the right direction?
thx

**Comment by grechaw** [comment]
Hi @lawrence-pressinnov we'll make it public at the end of this week, we're still getting it ready, stay tuned!

**Comment by grechaw** [comment]
Once we open up the new repo you can check it out @elky87 and we'll close this PR pending your review.

**Comment by grechaw** [comment]
Closing this now as these changes were incorporated into rdf4j

**Pull request closed without merging**